### PR TITLE
Rest api migration

### PR DESCRIPTION
**Pull request by Nassuel**
api_keys.txt is no longer tracked by using
```git update-index --assume-unchanged [path]```

Ran the script without any hiccups, but need to test further as code coverage is not the best atm

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -31,6 +31,10 @@ share/python-wheels/
 *.egg
 MANIFEST
 
+# Pyvenv stuff
```
Comment (Nassuel) [inline .gitignore:34]: Added this to ignore venv-generated files

**Comment by stacksjb** [comment]
I tested and it appears to work, but I'm still dealing with a few odd cases where it says an assignment is being added but no new tasks appear. Troubleshooting and will figure out further.  Once I validate it works I think we can merge.

**Comment by stacksjb** [comment]
I tested this and ran into a few issues. 

1) It encounters an error creating projects in Todoist if they don't already exist
`if courses_id_name_dict[course_id] not in todoist_project_dict:`

2) I ran into several errors until I figured out the updates you had made were to the canvas_to_python script and not easy_run.py (I haven't run the canvas_to_python in ages), so these updates didn't make any changes to easy_run (which also pulls from config.json instead of api_keys.txt). 

I think we should deprecate/remove the older script/files and just use the easy_run, what are your thoughts?

**Comment by stacksjb** [comment]
I confirmed your changes work successfully once I integrated the code changes in manually into the easy_run.py

I'll approve this PR and add you to the contributors page (thanks!) and then make the changes to merge those files 
(I also have some upstream merges and changed features I need to push down and include, so will have to resolve all of those as well).

**Review by stacksjb — APPROVED** [review]
Approved

**Comment by stacksjb** [comment]
Merged files for canvas_to_todoist.py. Will make edits for easy_run.py

**Comment by Nassuel** [comment]
> I think we should deprecate/remove the older script/files and just use the easy_run, what are your thoughts?

I agree. Anything on the contrary, @scottquach?

**Comment by stacksjb** [comment]
I've completed the merge, but need @scottquach to remove the three no longer used files so I we can resolve and merge the updates in from my branch.

My branch included changes to add priority, autosetup the config.json (file isn't needed to be present), add labels (one great change with the new API is it now handles Labels by name, so I don't need the lookup or hackery code to do ID lookup code)

I have long term plans to work on another version, but for the forseeable future will be adding several things to this version as I rely on it quite heavily.

Thanks!

**Pull request merged**